content/dev/blog/analyzing-so-data.Rmd

```diff
@@ -0,0 +1,308 @@
+```{r Illustrate problem columns, results='asis'}
+
+vars <- schema$Column[str_detect(schema$Question,
+    "Congratulations! You've just been put in charge of technical recruiting at Globex")]
+
+stack_df[1:10, ] %>%
+    select(one_of(vars)) %>%
+    knitr::kable()
+```
+
+A better solution would be to store the responses to this question in some kind
+of tabular structure which showed the relationship between them. For instance we
+could calculate a frequency table which showed the percentage of respondents who
+gave a variable a particular rating. We can do this locally in R with something
+like the following:
+
+```{r cat array mocking, cache = TRUE, results='asis'}
+
+catarray <- stack_df[, vars] %>%
+    gather(variable, rating) %>%
+    group_by(variable, rating) %>%
+    tally() %>%
+    filter(!is.na(rating)) %>%
+    mutate(n = scales::percent(n/sum(n))) %>%
+    spread(rating, n)
+catarray$variable <- str_replace(catarray$variable, "ImportantHiring", "")
+catarray[ ,c("variable", "Very important",
+    "Important", "Somewhat important", "Not very important", "Not at all important")] %>%
+    knitr::kable()
+```
+
+This picture is great for summarizing a set of related categorical variables,
+but it's a bit of a pain to produce, and is difficult to display alongside other
+variable summaries. Since this is such a common task, Crunch has a special data
+type called a "categorical array" for working with related categorical
+variables.
+
+```{r, cache = TRUE}
+ds$importanceHiringArray <- makeArray(ds[, vars], name = "Importance to Hiring")
+knitr::include_graphics('./images/catarray_screen.png')
+```
+
+Another common survey response data type can be found in Q350
+
+>   Q350. When it comes to compensation and benefits, other than base salary,
+>   which f the following are the most important to you? Please choose no more
+>   than X.
+>
+>   * Stock options
+>   * Retirement/pension contributions
+>   * Opportunity for an annual bonus
+>   * The number of annual days off (vacation, holidays, etc.)
+>   * Employer match of charitable contributions
+>   * Health benefits
+>   * Employer purchase of high-quality equipment (workstation, monitor, etc.)
+>   * Private office
+>   * Employer sponsorship of professional development (e.g. conference attendance, course enrollment)
+>   * Employer sponsorship of education (e.g. tuition reimbursement)
+>   * Long-term leave policies (e.g. parental leave, sabbatical)
+>   * Number of expected work hours each week \* The ability to work from home
+>   * Child or elder care benefits
+
+This is a multiple response question which allows respondents to select a number
+of factors which they consider important. The responses are stored in semi-colon
+delimited lists:
+
+```{r, cache = TRUE, results='asis'}
+stack_df[1:10, c("Respondent", "ImportantBenefits")] %>%
+    knitr::kable()
+```
+
+The first task is to split up the responses into columns which show whether a
```

Comment:
Probably should do https://github.com/Crunch-io/rcrunch/issues/149 and then you can collapse this section and forego the gory details of how we parsed semicolons and such.